src/coreclr/jit/emitarm64.cpp

```diff
@@ -8642,26 +8665,31 @@ void emitter::emitIns_R_R_R(
             assert(isFloatReg(reg1));
             assert(isLowPredicateRegister(reg2));
             assert(isVectorRegister(reg3));
-            assert(insOptsScalableWithSimdScalar(opt));
+            assert(insOptsScalable(opt));
+            assert(sopt == INS_SCALABLE_OPTS_NONE);
             fmt = IF_SVE_AF_3A;
             break;
 
         case INS_sve_andqv:
         case INS_sve_eorqv:
         case INS_sve_orqv:
-            unreached(); // TODO-SVE: Not yet supported.
             assert(isVectorRegister(reg1));
             assert(isLowPredicateRegister(reg2));
             assert(isVectorRegister(reg3));
-            assert(insOptsScalableWithSimdVector(opt));
+            assert(insOptsScalable(opt));
+            assert(sopt == INS_SCALABLE_OPTS_NONE);
```

Comment:
shouldn't this be `INS_SCALABLE_OPTS_WITH_SIMD_SCALAR`?

Reply:
likewise for few below where previously we had `assert(insOptsScalableWithSimdVector(opt))`

Reply:
There is only a single encoding for `andqv`. It always has a destination of a SIMD vector. This is the same for _every_ instruction that uses a SIMD vector. In all these cases, the _only_ use of setting `sopt` would be for an assert check. Removing the setting of `sopt` meant there were no uses of `_WITH_SIMD_VECTOR` and it could be removed.

Meanwhile, have a look above at `andv`. This always has a destination of a SIMD _scalar_. This is the same for some other instructions (eg `saddv`), but there are some instructions which can optionally have a destination of SVE Vector or SIMD scalar (eg `clasta` or `mov`). We have a choice here:

- Always pass `_WITH_SIMD_SCALAR` to any instruction with a SIMD scalar.
- Only pass `_WITH_SIMD_SCALAR` to instructions where there is an option (`clasta/mov`). For instructions without a option (`andv`) do not use an `sopt`.

The PR goes with the second option. Why?....

Finally, have a look at `movprfx`. This can have optionally have zero or merge predicate, and uses `_PREDICATE_MERGE` to distinguish. If we were to insist on always passing `_WITH_SIMD_SCALAR` to `andv`, then technically we'd have to pass `_PREDICATE_MERGE` to every instruction that uses a predicate merge. Which is a lot of instructions, and becomes a pain to ensure we're always using it.

Reply:
That makes sense. I am wondering if we can capture this in a comment somewhere so the other developers can follow and reason about this design.

Reply:
I added [this comment ](https://github.com/dotnet/runtime/pull/96692/files#diff-30bb3a9b91e000890df4038b7deb5255190a08aeb82ed31610206389b8fab73aR306:~:text=//%20When%20a%20single%20instruction%20has%20different%20encodings%20variants%2C%20this%20is%20used) for the insScalableOpts definition in instr.h